File: Project/Boundaries/Endpoint.py
from datetime import datetime


class Endpoint:
    def __init__(self, endpoint_id: str, policy_id: str, hostname: str, ip_address: str, status: str,
                 last_communication=datetime.now()):
        self.id = endpoint_id
        self.hostname = hostname
        self.IPAddress = ip_address
        self.policyId = policy_id
        self.status = status
        self.lastCommunication = last_communication

    def __str__(self):
        return f"ןג:{self.id}, " \
               f"IPAddress:{self.IPAddress}, " \
               f"Hostname:{self.hostname}, " \
               f"PolicyId:{self.policyId}" \
               f"Status:{self.status}" \
               f"LastCommunication:{self.lastCommunication}"

File: Web/src/Components/Endpoints/EndpointClasses/Endpoint.js
import { makeObservable, observable, runInAction } from 'mobx';
import axios from 'axios';

export default class Policy {
    id = 0;
    policyId = 0;
    name = '';
    ipAddress = [];
    policyName = '';
    status = '';
    lastCommunication = '';

    constructor({
        id,
        hostname,
        IPAddress,
        policyId,
        status,
        lastCommunication,
    }) {
        makeObservable(this, {
            id: observable,
            policyId: observable,
            name: observable,
            ipAddress: observable,
            policyName: observable,
            status: observable,
            lastCommunication: observable,
        });
        this.id = id;
        this.policyId = policyId;
        this.name = hostname;
        this.ipAddress = IPAddress;
        this.status = status;
        this.lastCommunication = lastCommunication;
    }

    updatePolicy = ({ policyId }) => {
        return axios
            .put(`/endpoint/update/${this.id}`, { policyId })
            .then((response) => {
                if (response.status === 200) {
                    runInAction(() => {
                        this.policyId = policyId;
                    });
                    return true;
                }
                return false;
            })
            .catch((error) => {
                console.log(error);
                return false;
            });
    };
}
